### ISSUE TITLE
Explain why we can't have an Item property when we declare an indexer property with its default name

### ISSUE THREAD

**Issue by SpiritBob** [dotnet/docs#19510]
Nowhere in the official docs does it state how exactly an indexer is born to fruition with the compiler.

Because that's not mentioned, one would greatly be confused when they attempt to create a property named "Item", when they have a named indexer property defined with the default index name (Item).

It would be wise to share with the user base that the way the compiler handles an indexer, is that it creates a property/methods for us, which can be used by languages not supporting indexers directly, that is:

If we have an object `SomeObject`, and it has an indexer with a single `int` argument, in C# if we would write the following: `SomeObject a = new SomeObject(); a[5];`, a language not supporting this would write something like this instead `SomeObject a = new SomeObject(); a.get_items(5);`

TL;DR: Any magic under the hood is worth pointing out.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5323b38b-8dc5-8f50-f5e7-b58ca0e88ad7
* Version Independent ID: 3d9a83cd-98e3-fc6e-727b-32c8e9228c69
* Content: [Using Indexers - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/using-indexers)
* Content Source: [docs/csharp/programming-guide/indexers/using-indexers.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/indexers/using-indexers.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by IEvangelist** [comment]
Hi 👋 @SpiritBob 
Thank you so much for posting this issue. We really appreciate it.

This is actually called out in the docs under the [**Remarks** section](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/using-indexers#remarks), with a code example too. Here is a [SharpLab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKBQBgAIUMAWAblwOIwDoAlAVwDsYBLMOWgYQHswADmwA2iAMqIAbmwDGcKJRxUAzMTSEx/OAHlgAKzgyYAHgAqAPlwBvXITuEA9A8IBBQgIS8BiGAE9CzBCcACaEAJLwYLb2KKqm4ZGEVoQA5nAw5IRQ6ZkAvtF2BYQA2mHMwXAAHogAckFwABQARKYAFnARcGBNAJQAukWxhPEwrWxQxWyshFMVlQM49kmFi0v2aTCEALzmhAD6bJETs1V9imv22Zs7+4ddx+Wn24SSEMKMcOf2+asrSwhwCDBXjMYT+AAy4xMFluR2ezDgAHdCJDYGZzA0eoofio1IQ+MwoIxOAhrINVASiSTMUUbL81q8EIRePp4UiNFpdAYjMYACIQeCmDhwDFYopLFl6YoAdj6z35guFtBqvERXwujJebw+z0lMrORR+uSAA) example of the code as compiled with name collision prevention.

In other words, you could rename the default `Item` property.

```csharp
using System;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

public class SomeObject<T>
{
    // A property named Item
    public T Item { get; set; }
    
    [IndexerName("TheItem")]
    public T this[int index]
    {  
        get => _items[index];
        set => _items[index] = value;
    }
    
    readonly List<T> _items = new List<T>();
}

public class Consumer
{
    public Consumer()
    {
        var obj = new SomeObject<DateTime>();
        obj[7] = DateTime.Now;
        var value = obj[7];
    }
}
```

Sometimes there is "compiler magic" that is difficult to document, this is actually covered in the doc. Does that help?

**Comment by SpiritBob** [comment]
Hello @IEvangelist , I appreciate all the Microsoft's team has done for us in providing a better documentation for future and current developers.

Prior to posting this issue I was aware that the default indexer name was pointed out in the docs. I just think that going a bit lower into the topic at hand would bring about additional clarity.

As it is written right now, a reader certainly understands that indexers can be named and renamed, but probably is not aware that a property named as an indexer will throw them a compile-time error, if that makes sense?

**Comment by IEvangelist** [comment]
That makes sense @SpiritBob - I'll add a bit of clarity in the doc. Thanks
